protostar/commands/test/test_collector.py

```diff
@@ -151,3 +170,156 @@ def _preprocess_contract(self, file_path: Path) -> StarknetPreprocessedProgram:
+        )
+
+        test_suites = self._build_test_suites_from_target_test_cases_info_mapping(
+            target_test_cases_info_mapping
+        )
+
+        return TestCollector.Result(
+            test_suites=test_suites,
+        )
+
+    def build_test_cases_mapping(
+        self,
+        test_suite_and_case_globs: Set[TestSuiteAndCaseGlob],
+    ) -> TestCasesMapping:
+        results: TestCasesMapping = {}
+
+        for test_suite_and_case_glob in test_suite_and_case_globs:
+            # split globs
+            test_suite_glob: Optional[TestSuiteGlob] = test_suite_and_case_glob
+            test_case_glob: Optional[TestCaseGlob] = None
+            if "::" in test_suite_and_case_glob:
+                (test_suite_glob, test_case_glob) = test_suite_and_case_glob.split("::")
+            test_suite_glob = test_suite_glob or self._default_test_suite_glob
+
+            if not test_suite_glob:
+                continue
+
+            # get filepaths and create the mapping
+            test_suite_filepaths = self._find_test_suite_filepaths_from_glob(
+                test_suite_glob
+            )
+            for test_suite_filepath in test_suite_filepaths:
+                test_case_globs = results.setdefault(test_suite_filepath, set())
+                if test_case_glob:
+                    test_case_globs.add(test_case_glob)
+
+        return results
+
+    # pylint: disable=no-self-use
+    def build_target_test_cases_info_mapping(
+        self,
+        test_cases_mapping: TestCasesMapping,
+        ignored_test_cases_mapping: TestCasesMapping,
+    ) -> TargetTestCasesInfoMapping:
+        # filter out test suites, which shouldn't be parsed
+        filtered_test_cases_mapping = test_cases_mapping
+        for ignored_target_path in ignored_test_cases_mapping:
+            if (
+                len(ignored_test_cases_mapping[ignored_target_path]) == 0
+                and ignored_target_path in filtered_test_cases_mapping
+            ):
+                del filtered_test_cases_mapping[ignored_target_path]
+
+        # create the mapping
+        result: TargetTestCasesInfoMapping = {}
+        for test_suite_filepath in filtered_test_cases_mapping:
+            target_test_cases_info = result.setdefault(
+                test_suite_filepath,
+                TargetTestCasesInfo(test_case_globs=set(), ignored_test_cases=set()),
+            )
+            target_test_cases_info.test_case_globs = filtered_test_cases_mapping[
+                test_suite_filepath
+            ]
+            if test_suite_filepath in ignored_test_cases_mapping:
+                target_test_cases_info.ignored_test_cases = ignored_test_cases_mapping[
+                    test_suite_filepath
+                ]
+
+        return result
+
+    def _find_test_suite_filepaths_from_glob(self, test_suite_glob: str) -> Set[str]:
+        results: Set[str] = set()
+        matches = glob(test_suite_glob, recursive=True)
```

Comment:
```suggestion
        matches = map(Path, glob(test_suite_glob, recursive=True))
```

Reply:
I do the mapping in the loop below to avoid running the loop twice.